contracts/Manager.sol

```diff
@@ -722,97 +429,179 @@ contract Manager is UUPSOwnableUpgradeable, UsingRegistryUpgradeable {
-        if (!activeGroups.remove(group)) {
-            revert GroupNotActive(group);
+    function distributeVotes(uint256 votes, address strategy)
+        private
+        returns (address[] memory finalGroups, uint256[] memory finalVotes)
+    {
+        if (strategy != address(0)) {
+            finalGroups = new address[](1);
+            finalVotes = new uint256[](1);
+            finalGroups[0] = strategy;
+            finalVotes[0] = votes;
+        } else {
+            (finalGroups, finalVotes) = defaultStrategy.generateGroupVotesToDistributeTo(votes);
         }
 
-        emit GroupDeprecated(group);
+        return (finalGroups, finalVotes);
+    }
 
-        if (account.getCeloForGroup(group) > 0) {
-            if (!deprecatedGroups.add(group)) {
-                revert FailedToAddDeprecatedGroup(group);
-            }
+    /**
+     * @notice Distributes withdrawals according to chosend strategy.
+     * @param stCeloAmount The amount of stCelo to be withdrawn.
+     * @param beneficiary The address that should end up receiving the withdrawn
+     * CELO.
+     **/
+    function distributeWithdrawals(uint256 stCeloAmount, address beneficiary) private {
+        uint256 withdrawal = toCelo(stCeloAmount);
+        if (withdrawal == 0) {
+            revert ZeroWithdrawal();
+        }
+
+        address strategy = _checkAndUpdateStrategy(beneficiary, strategies[beneficiary]);
+
+        address[] memory groupsWithdrawn;
+        uint256[] memory withdrawalsPerGroup;
+
+        if (strategy != address(0)) {
+            (groupsWithdrawn, withdrawalsPerGroup) = specificGroupStrategy
+                .calculateAndUpdateForWithdrawal(strategy, withdrawal, stCeloAmount);
         } else {
-            emit GroupRemoved(group);
+            (groupsWithdrawn, withdrawalsPerGroup) = defaultStrategy
+                .calculateAndUpdateForWithdrawal(withdrawal);
         }
+
+        account.scheduleWithdrawals(beneficiary, groupsWithdrawn, withdrawalsPerGroup);
     }
 
     /**
-     * @notice Checks if a group member is elected.
-     * @param groupMember The member of the group to check election status for.
-     * @return Whether or not the group member is elected.
+     * @notice Checks if strategy was deprecated. Deprecated strategy is reverted to default.
+     * Updates the strategies.
+     * @param accountAddress The account.
+     * @param strategy The strategy.
+     * @return Up to date strategy
      */
-    function isGroupMemberElected(address groupMember) private view returns (bool) {
-        IElection election = getElection();
-
-        address[] memory electedValidatorSigners = election.electValidatorSigners();
+    function _checkAndUpdateStrategy(address accountAddress, address strategy)
+        private
+        returns (address)
+    {
+        address checkedStrategy = _checkStrategy(strategy);
+        if (checkedStrategy != strategy) {
+            strategies[accountAddress] = checkedStrategy;
```

Comment:
Not sure if we should ever force updates to the strategy? Shouldn't be necessary if we follow the "a specific strategy that's targeting a currently ineligible group will at the time vote according to the default strategy, but will change back to the group if it becomes eligible" idea.
What are the cases when _checkStrategy would return a different strategy (i.e., when would isSpecificGroupStrategy fail)?

Reply:
This will be addressed as part of overflow task